libraries/apollo-normalized-cache-api-incubating/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/MemoryCache.kt

```diff
@@ -18,141 +20,242 @@ import kotlin.reflect.KClass
  * (there is no any sort of GC that runs in the background).
  */
 class MemoryCache(
+    private val nextCache: NormalizedCache? = null,
     private val maxSizeBytes: Int = Int.MAX_VALUE,
     private val expireAfterMillis: Long = -1,
-) : NormalizedCache() {
-  private val lock = Lock()
+) : OptimisticNormalizedCache {
+  // A lock is only needed if there is a nextCache
+  private val lock = nextCache?.let { Lock() }
```

Comment:
Do we still need our own `Lock`? Atomic fu says: ["JVM it is a typealias to the later class, erased on JS."](https://github.com/Kotlin/kotlinx-atomicfu?tab=readme-ov-file#locks)

Reply:
I remember that discussion now! 👍